lib/Sema/DerivedConformanceCodable.cpp

```diff
@@ -300,9 +300,17 @@ static CodingKeysValidity hasValidCodingKeysEnum(TypeChecker &tc,
   if (!tc.conformsToProtocol(codingKeysType, codingKeyProto,
```

Comment:
Do you need this part at all?

```
if (isa<TypeAliasDecl>(codingKeysTypeDecl))
     codingKeysTypeDecl = codingKeysType->getAnyNominal();
```

You're not using codingKeysDecl below for anything, except to get its location. So why not always use the location of the original type to diagnose, even if it was a type alias?

Reply:
See also the check on line 318; we confirm that the decl itself is indeed an enum